plugins/modules/rabbitmq_exchange.py

```diff
@@ -192,11 +199,20 @@ def main():
             result['changed'] = True
             module.exit_json(**result)
         else:
-            module.fail_json(
-                msg="Error creating exchange",
-                status=r.status_code,
-                details=r.text
-            )
+            rjson = r.json()
+            if (rjson['reason'].startswith('unknown exchange type')):
+                module.fail_json(
+                    msg=("Error creating exchange. You may need to enable the '%s' plugin for exchange type %s" %
+                         (exchange_plugins[module.params['exchange_type']], module.params['exchange_type'])),
```

Comment:
Maybe exchange_plugins should be a defaultdict, or we need to catch KeyError here for exchange types that we don't know. That should make it easier to test adding a new type without adding it everywhere.

Reply:
I've added a try / except KeyError, however, I don't think it will ever be triggered (except during future development) due to the arg spec definition.